contracts/teller-network/SellerStore.sol

```diff
@@ -39,19 +39,27 @@ contract SellerStore is Ownable {
     );
 
     struct Seller {
-        address asset;
         address statusContractCode;
         string location;
-        string currency;
         string username;
+    }
+
+    struct Offer {
+        address asset;
+        string currency;
+        uint8 margin;
         PaymenMethods[] paymentMethods;
         MarketType marketType;
-        uint8 margin;
     }
 
     address public license;
     Seller[] public sellers;
-    mapping(address => mapping (uint256 => bool)) public ownerToSellers;
+    Offer[] public offers;
+    mapping(address => bool) public sellerWhitelist;
+    mapping(address => uint256) public addressToSeller;
+
+    mapping(address => mapping (uint256 => bool)) public offerWhitelist;
+    mapping(address => uint256[]) public addressToOffers;
```

Comment:
Not sure what the whitelists are for? If we have a mapping with the address pointing to the offerID, can't we just check for the existence there and not have a whitelist? I probably just don,t know something about solidity here.

Reply:
I'll try to explain what I know and @richard-ramos can correct me if I am wrong:

if you have a mapping: `mapping(address => uint256) foo`
and access `foo[msg.sender]` if there is nothing in the mapping it will return the default value which will be 0.
However 0 is a valid ID.
So having a white list (mapping with boolean) I can be sure when it is true or false by setting it explicitly to true and the default value being false

Makes sense?

Reply:
Yeah makes sense. Can't we just start IDs at 1 then?

Reply:
IDs are not controlled by me but are indexes of array

I think that would add a lot of overhead:
* Need to add an empty seller/offer in array by default to avoid id 0
* Rewrite public getter to ignore this id 0
...
And probably others.
I have seen this technique quite a lot and seems to be the suggested way of doing this kind of manipulation.
However I might be wrong what do you think @richard-ramos

Reply:
Ah ok. Will wait on to see what Richard says, but what you're saying makes a lot of sense

Reply:
If to be a seller, we require a license, isn't it possible to remove the need for a sellerWhitelist by calling `isLicenseOwner` from the license contract?

Reply:
You can be a license owner but not be a seller yet

Reply:
how would you do that?

Reply:
I'm a little confused due to this code here:
```
if (!sellerWhitelist[msg.sender]) {
            Seller memory seller = Seller(_statusContractCode, _location, _username);
            uint256 sellerId = sellers.push(seller) - 1;
            addressToSeller[msg.sender] = sellerId;
            sellerWhitelist[msg.sender] = true;
        }
```
According to this: ` addressToSeller[msg.sender] `  will always contain the latest `sellerId` right? Is it possible that I as a seller can update different seller records except the latest, because in `update` we are always updating the latest seller data record.

Reply:
Or another question:
can a Seller have more than one  
```
struct Seller {
        address statusContractCode;
        string location;
        string username;
    }
``` 
?

Reply:
Don't think so, that's 1 to 1.

Reply:
It is a 1-1 so one address can only have 1 seller this is why I do:
```
if (!sellerWhitelist[msg.sender]) {
  Seller memory seller = Seller(_statusContractCode, _location, _username);
  uint256 sellerId = sellers.push(seller) - 1;
  addressToSeller[msg.sender] = sellerId;
  sellerWhitelist[msg.sender] = true;
}
```

Reply:
I think in this case, we can remove the need for an array and withelist like this.   i.e.: 

    
```
mapping(address => Seller) public sellers;


function add(
        address _asset,
        address _statusContractCode,
        string memory _location,
        string memory _currency,
        string memory _username,
        PaymenMethods[] memory _paymentMethods,
        MarketType _marketType,
        uint8 _margin
    ) public {
        require(License(license).isLicenseOwner(msg.sender), "Not a license owner");
        require(_margin <= 100, "Margin too high");

        
        sellers[msg.sender].statusContractCode = _statusContractCode;
        sellers[msg.sender].location = _location;
        sellers[msg.sender].username = _username;

        
        Offer memory offer = Offer(_asset, _currency, _margin, _paymentMethods, _marketType);
        uint256 offerId = offers.push(offer) - 1;
        offerWhitelist[msg.sender][offerId] = true;
        addressToOffers[msg.sender].push(offerId);

        emit Added(msg.sender, offerId, _asset, _statusContractCode, _location, _currency, _username, _paymentMethods, _marketType, _margin);
    }
```
what do you think?

Reply:
I wanted to do that but then how do you access a list of seller?

Reply:
You're right!

The more I look at this, the greater the need I see to have this information stored offchain instead of using a contract. Filtering the data will be difficult using this model.